### PR TITLE
Docs: Expanding the descriptions of updateData(), loadData() and their hooks

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR:
- Expands the descriptions of the `updateData()`, `loadData()` methods and their hooks
- Marks out the desired structure and style of all hooks' descriptions (to be rewritten in the future)

[skip changelog]

### CODE REVIEW

**Review by warpech — CHANGES_REQUESTED** [review]
Thanks for the improvements. I have a few comments, though.

File: handsontable/src/pluginHooks.js [hunk 1]
```diff
@@ -297,34 +298,48 @@ const REGISTERED_HOOKS = [
   'afterInit',
 
   /**
-   * Fired after new data is loaded (by `loadData` or `updateSettings` method) into the data source array.
+   * Fired after the [`loadData()`](@/api/core.md#loaddata) method,
+   * or the [`updateSettings()`](@/api/core.md#updatesettings) method,
+   * modifies Handsontable's [`data`](@/api/options.md#data).
```
[the remark this answers is not in the document]
Comment (kirszenbaum) [reply handsontable/src/pluginHooks.js:303]: @warpech
Re: 1.
Generally, it's good practice to avoid past tense and passive voice (unless necessary). But we don't have style guide rules set for such descriptions yet, so, for consistency with the other hook descriptions, I'd go for:
    ```
    Fired after Handsontable's `data` gets modified by the `loadData()` method or the `updateSettings()` method.
    ```

Re: 2.
> In the description of updateSettings() I can't see that it fires afterSetData, but the name of that method is mentioned here

As I understand it, `updateSettings()` fires the `afterLoadData` and `beforeLoadData` hooks **only** when it modifies the `data` configuration option (@jansiegel pls let me know if I'm wrong) – that's why those hooks were not listed under `updateSettings()` before. We can add them, but wouldn't it affect other hooks as well? (e.g. if you use `updateSettings()` to change the `collapsibleColumns` option, it would fire the `afterColumnCollapse`, right?)

Re: 3.
As discussed, I removed `setData()` from the docs.

File: handsontable/src/pluginHooks.js [hunk 2]
```diff
@@ -297,34 +298,48 @@ const REGISTERED_HOOKS = [
...
+   * @param {boolean} initialLoad A flag that indicates whether the data was loaded at Handsontable's initialization (`true`) or later (`false`)
+   * @param {string} source The source of the call
    */
   'afterUpdateData',
 
   /**
-   * Fired after new data is loaded (by `setData` or `updateSettings` method) into the data source array.
+   * Fired after the [`setData()`](@/api/core.md#setdata) method,
+   * or the [`updateSettings()`](@/api/core.md#updatesettings) method,
+   * modifies Handsontable's [`data`](@/api/options.md#data).
```
[the remark this answers is not in the document]
Comment (kirszenbaum) [reply handsontable/src/pluginHooks.js]: Replied here: https://github.com/handsontable/handsontable/pull/9084#discussion_r782068197

File: handsontable/src/pluginHooks.js [hunk 3]
```diff
@@ -798,38 +813,52 @@ const REGISTERED_HOOKS = [
   'beforeInitWalkontable',
 
   /**
-   * Fired before new data is loaded (by `loadData` or `updateSettings` method) into the data source array.
+   * Fired before the [`loadData()`](@/api/core.md#loaddata) method,
+   * or the [`updateSettings()`](@/api/core.md#updatesettings) method,
+   * modifies Handsontable's [`data`](@/api/options.md#data).
```
[the remark this answers is not in the document]
Comment (kirszenbaum) [reply handsontable/src/pluginHooks.js]: Replied here: https://github.com/handsontable/handsontable/pull/9084#discussion_r782068197

File: handsontable/src/pluginHooks.js [hunk 4]
```diff
@@ -798,38 +813,52 @@ const REGISTERED_HOOKS = [
...
+   * @param {string} source The source of the call
+   * @returns {Array} The returned array will be used as Handsontable's new dataset.
    */
   'beforeUpdateData',
 
   /**
-   * Fired before new data is loaded (by `setData` or `updateSettings` method) into the data source array.
+   * Fired before the [`setData()`](@/api/core.md#setdata) method,
+   * or the [`updateSettings()`](@/api/core.md#updatesettings) method,
+   * modifies Handsontable's [`data`](@/api/options.md#data).
```
[the remark this answers is not in the document]
Comment (kirszenbaum) [reply handsontable/src/pluginHooks.js]: Replied here: https://github.com/handsontable/handsontable/pull/9084#discussion_r782068197

File: handsontable/src/core.js [hunk 5]
```diff
@@ -94,7 +94,6 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
    * @type {HTMLElement}
```
[the remark this answers is not in the document]
Comment (kirszenbaum) [reply handsontable/src/core.js:94]: @warpech - I added a hint about using `updateData()` wherever `loadData()` is used in the docs: 85b111f

File: docs/11.1/api/hooks.md [hunk 6]
```diff
@@ -1662,21 +1648,26 @@ Fired by [TrimRows](@/api/trimRows.md) plugin after untrimming rows. This hook i
 
 ### afterUpdateData
   
-::: source-code-link https://github.com/handsontable/handsontable/blob/0472af66268f29ceb64d1f046b74a05149cffe8d/../handsontable/src/pluginHooks.js#L309
+::: source-code-link https://github.com/handsontable/handsontable/blob/5bef6a42994fcc395de546c0f8adbc297579650d/../handsontable/src/pluginHooks.js#L316
```
Comment (warpech) [inline docs/11.1/api/hooks.md:1651]: How is this tested? These links in the docs do not work, as reported https://github.com/handsontable/handsontable/issues/8948. We can fix this when a new person joins to help us work on the automations.

File: docs/11.1/api/core.md [hunk 7]
```diff
@@ -1415,16 +1415,23 @@ in the right way.
 
 _core.loadData(data, [source])_
 
-Loads new data to Handsontable. Loading new data resets the cell meta.
-Since 8.0.0 loading new data also resets states corresponding to rows and columns
-(for example, row/column sequence, column width, row height, frozen columns etc.).
+The `loadData()` method:
+- Replaces Handsontable's [`data`](@/api/options.md#data)
+- Resets cells' [configuration options](@/guides/getting-started/setting-options.md)
```
Comment (warpech) [inline docs/11.1/api/core.md:1420]: I need an explanation what does it mean that it resets the configuration options. Maybe this point should be removed?

The below point (about resetting states) is clear to me, but this one is not.

**Review by warpech — CHANGES_REQUESTED** [review]

File: docs/11.1/api/core.md [hunk 8]
```diff
@@ -1415,16 +1415,23 @@ in the right way.
...
+- Resets rows' and columns' states (e.g. [row height](@/guides/rows/row-height.md), [column width](@/guides/columns/column-width.md), frozen [rows](@/guides/rows/row-freezing.md) and [columns](@/guides/columns/column-freezing.md) etc.)
```
Comment (jansiegel) [inline docs/11.1/api/core.md:1421]: 1. That's generally true when manually setting the column widths/row heights using `manualaColumnResize`/`manualRowResize`, but if they're fixed in the config object during initialization, they'll stay the same after calling `loadData`. For example:
```
const hot = new Handsontable(container, {
      colWidths: [100, 20]
});
```

or 

```
const hot = new Handsontable(container, {
      manualColumnWidths: [100, 20]
});
```

2. I don't think it's correct in terms of freezing/fixing columns/rows. They seem to remain unchanged regardless of the method being used:
- `loadData`: https://jsfiddle.net/js_ziggle/sqtx87uc/
- `updateData`: https://jsfiddle.net/js_ziggle/fjL3yusx/

Basically, `loadData`  method resets:
- cell meta (cell styling, `readOnly` states, anything stored in the cell meta object)
- index mappers (mostly the row and column order)

File: docs/11.1/api/core.md [hunk 9]
```diff
@@ -2228,16 +2213,24 @@ any keyboard events.
 
 _core.updateData(data, [source])_
 
-Replaces the dataset with a new one. Unlike `setData` and the `loadData` methods, it doesn't reset the
-state of the table.
+The `updateData()` method:
+- Replaces Handsontable's [`data`](@/api/options.md#data)
+- Keeps cells' [configuration options](@/guides/getting-started/setting-options.md) unchanged
+- Keeps rows' and columns' states unchanged (e.g. [row height](@/guides/rows/row-height.md), [column width](@/guides/columns/column-width.md), frozen [rows](@/guides/rows/row-freezing.md) and [columns](@/guides/columns/column-freezing.md) etc.)
```
Comment (jansiegel) [inline docs/11.1/api/core.md:2219]: Same notes as with `loadData` above.

`updateData` doesn't clear:
- index mappers
- cell meta

Otherwise it does the same thing as `loadData`.

File: handsontable/src/core.js [hunk 10]
```diff
@@ -2019,14 +2018,22 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
   };
 
   /**
-   * Replaces the dataset with a new one. Unlike the `loadData` method, it doesn't reset the
-   * state of the table.
+   * The `updateData()` method:
+   * - Replaces Handsontable's [`data`](@/api/options.md#data)
+   * - Keeps cells' [configuration options](@/guides/getting-started/setting-options.md) unchanged
+   * - Keeps rows' and columns' states unchanged (e.g. [row height](@/guides/rows/row-height.md), [column width](@/guides/columns/column-width.md), frozen [rows](@/guides/rows/row-freezing.md) and [columns](@/guides/columns/column-freezing.md) etc.)
```
Comment (jansiegel) [inline handsontable/src/core.js:2024]: Same as the comments above.

File: handsontable/src/core.js [hunk 11]
```diff
@@ -2055,14 +2062,21 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
   };
 
   /**
-   * Loads new data to Handsontable. Loading new data resets the cell meta.
-   * Since 8.0.0 loading new data also resets states corresponding to rows and columns
-   * (for example, row/column sequence, column width, row height, frozen columns etc.).
+   * The `loadData()` method:
+   * - Replaces Handsontable's [`data`](@/api/options.md#data)
+   * - Resets cells' [configuration options](@/guides/getting-started/setting-options.md)
+   * - Resets rows' and columns' states (e.g. [row height](@/guides/rows/row-height.md), [column width](@/guides/columns/column-width.md), frozen [rows](@/guides/rows/row-freezing.md) and [columns](@/guides/columns/column-freezing.md) etc.)
```
Comment (jansiegel) [inline handsontable/src/core.js:2068]: Same as the comments above.

**Review by jansiegel — CHANGES_REQUESTED** [review]

File: docs/11.1/api/hooks.md (hunk 6 above)
Comment (kirszenbaum) [reply docs/11.1/api/hooks.md:1651]: @warpech - should I manually update the links for now?

File: docs/11.1/api/core.md (hunk 7 above)
Comment (kirszenbaum) [reply docs/11.1/api/core.md]: @warpech - please have a look now at the updated version: 64e6f14

File: docs/11.1/api/core.md (hunk 8 above)
Comment (kirszenbaum) [reply docs/11.1/api/core.md]: @jansiegel - please have a look now at the updated version: 64e6f14

File: docs/11.1/api/core.md (hunk 9 above)
Comment (kirszenbaum) [reply docs/11.1/api/core.md]: @jansiegel - please have a look now at the updated version: 64e6f14

File: handsontable/src/core.js (hunk 11 above)
Comment (kirszenbaum) [reply handsontable/src/core.js]: @jansiegel - please have a look now at the updated version: 64e6f14

**Review by jansiegel — APPROVED** [review]
LGTM

File: docs/11.1/api/hooks.md (hunk 6 above)
Comment (warpech) [reply docs/11.1/api/hooks.md:1651]: I don't know, I think it doesn't matter until #8948 is fixed.

**Review by warpech — APPROVED** [review]
